openfe/tests/protocols/test_openmm_equil_rfe_protocols.py

```diff
@@ -786,3 +788,57 @@ def test_constraints(tyk2_xml, tyk2_reference_xml):
             assert a.get('p1') == b.get('p1')
             assert a.get('p2') == b.get('p2')
             assert float(a.get('d')) == pytest.approx(float(b.get('d')))
+
+
+@pytest.fixture
+def available_platforms() -> set[str]:
+    return {Platform.getPlatform(i).getName() for i in range(Platform.getNumPlatforms())}
+
+
+@pytest.fixture
+def set_openmm_threads_1():
+    # for vacuum sims, we want to limit threads to one
+    # this fixture sets OPENMM_CPU_THREADS='1' for a single test, then reverts to previously held value
+    previous: str | None = os.environ.get('OPENMM_CPU_THREADS')
+
+    try:
+        os.environ['OPENMM_CPU_THREADS'] = '1'
+        yield
+    finally:
+        if previous is None:
+            del os.environ['OPENMM_CPU_THREADS']
+        else:
+            os.environ['OPENMM_CPU_THREADS'] = previous
+
+
+@pytest.mark.flaky(reruns=3)  # pytest-rerunfailures; we can get bad minimisation
+@pytest.mark.parametrize('platform', ['CPU', 'CUDA', 'OpenCL'])
+def test_openmm_run_engine(benzene_vacuum_system, platform, available_platforms,
+                           set_openmm_threads_1, tmpdir):
+    if os.environ['CI'] and os.environ['RUNNER_OS'] == 'macOS' and platform == 'OpenCL':
```

Comment:
We should probably just not allow OpenCL tbh, it's a wildly untested platform

Reply:
Is it that OpenCL will never work?

If we don't want to use OpenCL ever,  raise an issue and implement that in the Protocol if that's the case.

Reply:
For macOS I belive it will _never_ work - that's an OS specific thing (OpenCL got deprecated a long while ago).

OpenCL is generally untested - we _should_ test it at some point, but I just don't have the capacity to do so.